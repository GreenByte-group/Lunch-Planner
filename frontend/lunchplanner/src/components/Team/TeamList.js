import React from "react"
import axios from "axios"

import {HOST} from "../../Config"
import Team from "./Team";
import List from "material-ui/List";
import {withStyles} from "material-ui/styles/index";
import {Link} from "react-router-dom";
import FloatingActionButton from "../FloatingActionButton";

const styles = {
    root: {
        height: '100%',
        overflow: 'hidden',
    },
    list: {
        padding: 0,
    },
};

class TeamList extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            teams: [],
            search:null,
        }
    }

    componentDidMount() {
        this.setState({
            search: this.props.search,
        });

        let url;
        if(this.props.search)
            url = HOST + "/event/search/" + this.props.search;
        else
            url = HOST + "/event";

        axios.get(url)
            .then((response) => {
                this.setState({
                    teams: response.data,
                })
            })
    }

    render() {
        const { classes } = this.props;
        let teams = this.state.teams;
        return (
            <div className={classes.root}>
                <List className={classes.list}>
                    {teams.map((listValue)=>{
                        return <Team name={listValue.eventName}
                                      id={listValue.eventId}
                                     member={listValue.invitations}
                        />;
                    })}
                </List>
                <Link to="/team/create"><FloatingActionButton /></Link>
            </div>

        );
    }
}

export default withStyles(styles)(TeamList);