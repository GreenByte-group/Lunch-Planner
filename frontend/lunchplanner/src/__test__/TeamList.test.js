import React from 'react';
import TeamList from '../components/Team/TeamList';
import renderer from 'react-test-renderer';
import {MemoryRouter} from "react-router-dom";

it ("renders correctly", ()=>{
    const tree = renderer.create(
        <MemoryRouter>
        <TeamList/>
        </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});