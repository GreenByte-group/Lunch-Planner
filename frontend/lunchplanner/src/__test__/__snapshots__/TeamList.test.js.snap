// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="TeamList-root-1"
>
  <ul
    className="MuiList-root-3 MuiList-padding-4 TeamList-list-2"
  />
  <a
    href="/team/create"
    onClick={[Function]}
  >
    <button
      className="MuiButtonBase-root-25 MuiButton-root-9 MuiButton-raised-15 MuiButton-fab-20 MuiButton-raisedSecondary-17 FloatingActionButton-whiteSymbol-7 FloatingActionButton-floatingButton-8"
      disabled={false}
      onBlur={[Function]}
      onClick={undefined}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-10"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-28"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <g>
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </g>
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-34"
      />
    </button>
  </a>
</div>
`;
